docs/Targets.md

```diff
@@ -0,0 +1,341 @@
+
+```mermaid
+graph TD;
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+Blob3["Blob b3"]
+```
+
+#### GraphStorage
+
+The [`GraphStorage`](https://pkg.go.dev/oras.land/oras-go/v2/content#GraphStorage) interface extends [`Storage`](#storage) by adding support for predecessor finding. It provides the following functions:
+
+- `Fetch`
+- `Exists`
+- `Push`
+- **`Predecessors`**: Finds out the nodes directly pointing to a given node in the graph.
+
+For the [same graph](#storage), the `Predecessors` function would act as follows:
+
+```
+Predecessors(b0) == [m0]
+Predecessors(m0) == []
+```
+
+### Target
+
+The [`Target`](https://pkg.go.dev/oras.land/oras-go/v2#Target) interface represents a CAS with tagging capability. It provides the following functions:
+
+- `Fetch`
+- `Exists`
+- `Push`
+- **`Resolve`**: Resolves a tag string to a descriptor.
+- **`Tag`**: Associates a descriptor with a tag string.
+
+For example, consider a graph stored in a Target, where `m0` is associated with two tags, `"foo"` and `"bar"`:
+
+```mermaid
+graph TD;
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+
+TagFoo>"Tag: foo"]-.->M0
+TagBar>"Tag: bar"]-.->M0
+```
+
+The effects of the `Resolve` function would be:
+
+```
+  Resolve("foo") == m0
+  Resolve("bar") == m0
+Resolve("hello") == nil
+```
+
+If a new tag `"v1"` is added to `m0`, the graph would update as follows:
+
+```mermaid
+graph TD;
+
+M0["Manifest m0"]--config-->Blob0["Blob b0"]
+M0--layers-->Blob1["Blob b1"]
+M0--layers-->Blob2["Blob b2"]
+
+TagFoo>"Tag: foo"]-.->M0
+TagBar>"Tag: bar"]-.->M0
+TagV1>"Tag: v1"]-.->M0
+```
+
+### GraphTarget
+
+The [`GraphTarget`](https://pkg.go.dev/oras.land/oras-go/v2#GraphTarget) interface combines the capabilities of [`GraphStorage`](#graphstorage) and [`Target`](#target). It provides the following functions:
+
+- `Fetch`
+- `Exists`
+- `Push`
+- `Resolve`
+- `Tag`
+- `Predecessors`
+
+## Content Stores
+
+In `oras-go` v2, a content store is an implementation of the [`Target`](#target) interface—specifically, the [`GraphTarget`](#graphtarget) interface.
+
+The library provides four built-in content stores:
+
+- [Memory Store](#memory-store): Stores everything in memory.
+- [OCI Store](#oci-store): Stores content in the OCI-Image layout on the file system.
+- [File Store](#file-store): Stores location-addressable content on the file system.
+- [Repository Store](#repository-store): Communicates with remote artifact repositories (e.g. `ghcr.io`, `docker.io`).
+
+### Memory Store
+
+The memory store, available in the [`content/memory`](https://pkg.go.dev/oras.land/oras-go/v2/content/memory) package, stores all content in memory, where each blob's content is mapped to its corresponding descriptor.
+
+> [!TIP]
+> The memory store is often used for building and storing artifacts in memory before copying them to other stores, such as remote repositories.
```

Comment:
It would be good to mention that memory store is not persistent storage here.

Reply:
I think "in memory" already implies that it is not persistent🤔

Reply:
OK